src/Integration/src/RabbitMQ/Inbound/RabbitInboundChannelAdapter.cs

```diff
@@ -0,0 +1,263 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the Apache 2.0 License.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Extensions.Logging;
+using RabbitMQ.Client;
+using Steeltoe.Common.Contexts;
+using Steeltoe.Common.Retry;
+using Steeltoe.Common.Util;
+using Steeltoe.Integration.Endpoint;
+using Steeltoe.Integration.Rabbit.Support;
+using Steeltoe.Integration.Support;
+using Steeltoe.Messaging;
+using Steeltoe.Messaging.Converter;
+using Steeltoe.Messaging.RabbitMQ;
+using Steeltoe.Messaging.RabbitMQ.Batch;
+using Steeltoe.Messaging.RabbitMQ.Core;
+using Steeltoe.Messaging.RabbitMQ.Extensions;
+using Steeltoe.Messaging.RabbitMQ.Listener;
+using Steeltoe.Messaging.RabbitMQ.Support;
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using RabbitConverter = Steeltoe.Messaging.RabbitMQ.Support.Converter;
+
+namespace Steeltoe.Integration.Rabbit.Inbound
+{
+    public class RabbitInboundChannelAdapter : MessageProducerSupportEndpoint // ,IOrderlyShutdownCapable
+    {
+        private static readonly AsyncLocal<IAttributeAccessor> _attributesHolder = new AsyncLocal<IAttributeAccessor>();
+        private readonly ILogger _logger;
+
+        public RabbitInboundChannelAdapter(IApplicationContext context, AbstractMessageListenerContainer listenerContainer, ILogger logger = null)
+            : base(context, logger)
+        {
+            if (listenerContainer == null)
+            {
+                throw new ArgumentNullException(nameof(listenerContainer));
+            }
+
+            if (listenerContainer.MessageListener != null)
+            {
+                throw new ArgumentException("The listenerContainer provided to an RabbitMQ inbound Channel Adapter " +
+                            "must not have a MessageListener configured since the adapter " +
+                            "configure its own listener implementation.");
```

Comment:
```suggestion
                throw new ArgumentException("The listenerContainer provided to a RabbitMQ inbound Channel Adapter " +
                            "must not have a MessageListener configured since the adapter " +
                            "configures its own listener implementation.");
```